api/datastore/sql/sql.go

```diff
@@ -765,8 +793,8 @@ func (ds *SQLStore) PutFn(ctx context.Context, fn *models.Fn) (*models.Fn, error
 				config = :config,
 				annotations = :annotations,
 				updated_at = :updated_at
-			WHERE name=:name AND app_id=:app_id;`)
-		}
+			    WHERE id=:id AND app_id=:app_id;`)
```

Comment:
don't think this is right. Especially without the ID being validated as present in the validator.